compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -1349,7 +1349,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         }
                     }
                     GenericParamDefKind::Const { has_default } => {
-                        if !infer_args && has_default {
+                        if !infer_args
+                            && has_default
+                            && !tcx.has_attr(param.def_id, sym::rustc_host)
```

Comment:
How does this work for desugared params?

Reply:
desugared params get this attribute as well: [link](https://github.com/rust-lang/rust/pull/114545/files#diff-0a61b538a3cec072c76fecae4635af6a12ec3256860029ac70549c2aa53ab394L1433-R1470)